src/infra/alarm/template.yaml

```diff
@@ -750,3 +758,177 @@ Resources:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      ActionsEnabled: true
+      AlarmActions:
+        - !Ref AlertingSNSLowAlertNotificationTopicARN
+      AlarmDescription: !Sub "Over 5% of API Gateway requests are exceeding 2 seconds in a 10 minute period. ${RunBookURL}"
+      Namespace: AWS/ApiGateway
+      MetricName: Latency
+      ComparisonOperator: GreaterThanThreshold
+      ExtendedStatistic: p95
+      Threshold: 2000
+      EvaluationPeriods: 1
+      Period: 600
+      TreatMissingData: notBreaching
+      Dimensions:
+        - Name: ApiName
+          Value: !Ref PrivateRestApiName
+
+  ApiLatencyP99Taking2SecsOrLonger:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      ActionsEnabled: true
+      AlarmActions:
+        - !Ref AlertingSNSLowAlertNotificationTopicARN
+      AlarmDescription: "Over 80% of API Gateway requests are exceeding 2 seconds in a 10 minute period."
+      Namespace: AWS/ApiGateway
+      MetricName: Latency
+      ComparisonOperator: GreaterThanThreshold
+      ExtendedStatistic: p80
+      Threshold: 2000
+      EvaluationPeriods: 1
+      Period: 600
+      TreatMissingData: notBreaching
+      Dimensions:
+        - Name: ApiName
+          Value: !Ref PrivateRestApiName
+
+  ApiHttp5xxErrors:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      ActionsEnabled: true
+      AlarmActions:
+        - !Ref AlertingSNSLowAlertNotificationTopicARN
+      AlarmDescription: !Sub "Small proportion of 5XX errors on the API Gateway. ${RunBookURL}"
+      Threshold: 1
+      EvaluationPeriods: 5
+      DatapointsToAlarm: 2
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      TreatMissingData: notBreaching
+      Metrics:
+        - Id: errorThreshold
+          Label: errorThreshold
+          ReturnData: true
+          Expression: IF(invocations<1,0,errorPercentage)
+        - Id: invocations
+          ReturnData: false
+          MetricStat:
+            Metric:
+              Namespace: AWS/ApiGateway
+              MetricName: Count
+              Dimensions:
+                - Name: ApiName
+                  Value: !Ref PrivateRestApiName
+            Period: 60
+            Stat: Sum
+        - Id: errorPercentage
+          Label: errorPercentage
+          ReturnData: false
+          Expression: (error/invocations)*100
+        - Id: error
+          ReturnData: false
+          MetricStat:
+            Metric:
+              Namespace: AWS/ApiGateway
+              MetricName: 5XXError
+              Dimensions:
+                - Name: ApiName
+                  Value: !Ref PrivateRestApiName
+            Period: 60
+            Stat: Sum
+
+  ApiHttp5xxCriticalErrors:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      ActionsEnabled: true
+      AlarmActions:
+        - !Ref AlertingSNSHighAlertNotificationTopicARN
+      AlarmDescription: !Sub "Significant proportion of 5XX errors on the API Gateway. ${RunBookURL}"
+      Threshold: 80
```

Comment:
10 - I.e 10% of calls is a 500

Reply:
Does that facilitate a high alert? 80% is a high enough threshold to imply majority of requests are failing. With low volume, it would take one of out 10 5xx failures to raise a PagerDuty alert.

Reply:
If we change to 10%